### PR TITLE
Convert instructor gradebook to async

### PR DESCRIPTION
**Pull request by jonatanschroeder**
Also removed last instance of the callback version of `checkBelongs`. Since this name was ambiguous and needed to be renamed anyway, I also gave it a more meaningful name.

### CODE REVIEW

File: apps/prairielearn/src/lib/course.ts [hunk 1]
```diff
@@ -38,7 +37,6 @@ export async function checkBelongsAsync(
     throw new Error('access denied');
```
Comment (nwalters512) [inline apps/prairielearn/src/lib/course.ts:37]: Can we throw an `HttpStatusError` with 403 here?

**Review by nwalters512 — APPROVED** [review]
LGTM! I assume you tested this all locally?

**Comment by jonatanschroeder** [comment]
> LGTM! I assume you tested this all locally?

🎩 😉

File: apps/prairielearn/src/lib/course.ts (hunk 1 above)
Comment (jonatanschroeder) [reply apps/prairielearn/src/lib/course.ts]: Done in 8184cac3ab76b6f0060f9c0f2e622e534fb03ab8.

**Pull request merged**